packages/react-i18n/src/i18n.ts

```diff
@@ -299,6 +363,31 @@ export class I18n {
     return WEEK_START_DAYS.get(country) || DEFAULT_WEEK_START_DAY;
   }
 
+  // Intl.NumberFormat sometimes annotates the "curerncy symbol" with a country code.
+  // For example, in locale 'fr-FR', 'USD' is given the "symbol" of " $US".
+  // This method strips out the country-code annotation, if there is one.
+  // (So, for 'fr-FR' and 'USD', the return value would be " $").
+  //
+  // For other currencies, e.g. CHF and OMR, the "symbol" is the ISO currency code.
+  // In those cases, we return the full currency code without stripping the country.
+  getShortCurrencySymbol(currencyCode?: string) {
+    const currency = currencyCode || this.defaultCurrency || '';
+    const regionCode = currency.substring(0, 2);
+    const info = this.getCurrencySymbol(currency);
+    const shortSymbol = info.symbol.replace(regionCode, '');
+
+    if (shortSymbol.match(/[A-Z]/)) {
+      // It looks like we have an alphabetic code, not a symbol.  Return the full code.
```

Comment:
If we pull out the regex into a meaningful variable, we might be able to reduce the verbosity in these comments 

```
const alphabeticCode = /[A-Z]/;
return shortSymbol.match(alphabeticCode)
   ? info
   : {symbol: shortSymbol, prefixed: info.prefixed};
```

Reply:
Good idea--adjusting as suggested.